src/main/java/edu/ucsb/cs156/rec/controllers/RecommendationRequestController.java

```diff
@@ -0,0 +1,162 @@
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+
+import edu.ucsb.cs156.rec.entities.RecommendationRequest;
+import edu.ucsb.cs156.rec.errors.EntityNotFoundException;
+import edu.ucsb.cs156.rec.repositories.RecommendationRequestRepository;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.Parameter;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import jakarta.validation.Valid;
+import lombok.extern.slf4j.Slf4j;
+
+@Tag(name = "RecommendationRequests")
+@RequestMapping("/api/recommendationrequest")
+@RestController
+@Slf4j
+public class RecommendationRequestController extends ApiController {
+
+    @Autowired
+    RecommendationRequestRepository recommendationRequestRepository;
+
+    /**
+     * Get a single recommendation request by id
+     * 
+     * @param id the id of the recommendation request
+     * @return a recommendation request
+     */
+    @Operation(summary = "Get a single recommendation request")
+    @PreAuthorize("hasRole('ROLE_USER')")
+    @GetMapping("")
+    public RecommendationRequest getById(
+            @Parameter(name = "id") @RequestParam Long id) {
+        RecommendationRequest recommendationRequest = recommendationRequestRepository.findById(id)
+                .orElseThrow(() -> new EntityNotFoundException(RecommendationRequest.class, id));
+
+        return recommendationRequest;
+    }
+
+    /**
+     * Create a new recommendation request
+     * 
+     * @param professorName       the name of the professor
+     * @param professorEmail      the email of the professor
+     * @param requesterName       the name of the requester
+     * @param recommendationTypes the types of recommendations
+     * @param details             the details of the request
+     * @param status              the status of the request
+     * @param submissionDate      the submission date of the request
+     * @param completionDate      the completion date of the request
+     * @return the saved recommendation request
+     */
+
+    @Operation(summary = "Create a new recommendation request")
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
+    @PostMapping("/post")
+    public RecommendationRequest postRecommendationRequest(
+            @Parameter(name = "professorName") @RequestParam String professorName,
+            @Parameter(name = "professorEmail") @RequestParam String professorEmail,
+            @Parameter(name = "requesterName") @RequestParam String requesterName,
```

Comment:
Instead of getting the requester name and email from the post, get it from the "current user".

Otherwise, users can forge requests from other people.

I'll link you to some code that demonstrates how to do this.

Reply:
Here's an example.

Note that the user here is stored as a User object (which is the way that I think it probably should be done, rather than storing the email address and name, which is redundant, since those are both already associated with a user in the user's table).

But whether you make that change or not, you should follow the example here of getting the currentUser, and then setting the requester email and requester name from there, and not from parameters passed into the controlller.  This is more secure and will lead to better database integrity.

Reply:
https://github.com/ucsb-cs156/proj-courses/blob/4bdc2674508428d03cb2dcd2e9077d75cdaf5a3f/src/main/java/edu/ucsb/cs156/courses/controllers/PersonalSchedulesController.java#L79